src/test/java/org/opentripplanner/middleware/controllers/api/ManageLegTraversalTest.java

```diff
@@ -140,74 +253,99 @@ void canTrackLegWithoutDeviating() {
 
             ZonedDateTime currentTime = startOfTrip;
             double cumulativeTravelTime = 0;
-            for (Segment segment : segments) {
+            for (LegSegment legSegment : legSegments) {
                 trackedJourney.locations = List.of(
                     new TrackingLocation(
-                        segment.start.lat,
-                        segment.start.lon,
+                        legSegment.start.lat,
+                        legSegment.start.lon,
                         new Date(currentTime.toInstant().toEpochMilli())
                     )
                 );
-                assertEquals(
-                    ManageTripTracking.getTripStatus(trackedJourney, busStopToJusticeCenterItinerary).name(),
-                    TripStatus.ON_SCHEDULE.name()
-                );
-                cumulativeTravelTime += segment.timeInSegment;
+                TravelerPosition travelerPosition = new TravelerPosition(trackedJourney, busStopToJusticeCenterItinerary);
+                TripStatus tripStatus = TripStatus.getTripStatus(travelerPosition);
+                assertEquals(tripStatus.name(), TripStatus.ON_SCHEDULE.name());
+                cumulativeTravelTime += legSegment.timeInSegment;
                 currentTime = startOfTrip.plus(getSecondsToMilliseconds(cumulativeTravelTime), ChronoUnit.MILLIS);
             }
         }
     }
 
+    @Test
+    void canTrackTripTurnByTurn() {
+        ZonedDateTime startOfTrip = ZonedDateTime.ofInstant(
+            busStopToJusticeCenterItinerary.legs.get(0).startTime.toInstant(),
```

Comment:
Extract var for `busStopToJusticeCenterItinerary.legs.get(0)` and its first step.

Reply:
Sorry, I was meaning to remove this unless unit test.